source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -124,6 +126,55 @@ SplitRequestPtr SimpleRequest::create(Router& router,
   return request_ptr;
 }
 
+const std::string ErrorFaultRequest::FAULT_INJECTED_ERROR = "Fault Injected: Error";
+
+SplitRequestPtr ErrorFaultRequest::create(Router& router,
+                                          Common::Redis::RespValuePtr&& incoming_request,
+                                          SplitCallbacks& callbacks, CommandStats& command_stats,
+                                          TimeSource& time_source) {
+  std::unique_ptr<ErrorFaultRequest> request_ptr{
+      new ErrorFaultRequest(callbacks, command_stats, time_source)};
+
+  const auto route = router.upstreamPool(incoming_request->asArray()[1].asString());
```

Comment:
why is router needed here?

Reply:
Good call. I was thinking that if there's no route, we should not inject a fault but better to keep it simple.